common-hooks/tls-certificate/internal_tls.go

```
@@ -197,25 +258,60 @@ type certValues struct {
 
 // The certificate mapping "cert" -> "crt". We are migrating to "crt" naming for certificates
 // in values.
-func convCertToValues(cert certificate.Certificate) certValues {
+func convCertToValues(cert *certificate.Certificate) certValues {
 	return certValues{
 		CA:  string(cert.CA),
 		Crt: string(cert.Cert),
 		Key: string(cert.Key),
 	}
 }
 
-func generateNewSelfSignedTLS(input *pkg.HookInput, cn string, sans, usages []string) (certificate.Certificate, error) {
-	ca, err := certificate.GenerateCA(input.Logger,
-		cn,
-		certificate.WithKeyAlgo(keyAlgorithm),
-		certificate.WithKeySize(keySize),
-		certificate.WithCAExpiry(caExpiryDurationStr))
+var ErrCAIsInvalidOrOutdated = errors.New("ca is invalid or outdated")
+
+func getCommonCA(input *pkg.HookInput, valKey string) (*certificate.Authority, error) {
+	auth := new(certificate.Authority)
+
+	ca, ok := input.Values.GetOk(valKey)
+	if ok {
```

I think it would be better to return an error like `ErrNotFound` directly here if the ca is not retrieved.